bashlex/parser.py

```diff
@@ -13,7 +13,10 @@ def _partsspan(parts):
 )
 
 def handleNotImplemented(p, type):
-    if len(p) == 2:
+    if p.context._proceedonerror:
+        parts = _makeparts(p)
+        p[0] = ast.node(kind='unimplemented', parts=parts, pos=_partsspan(parts))
```

Comment:
i would just return at the end of this block, just to separate these two logically